modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -353,82 +359,131 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
+      dof_id_type base_rid = elem->get_extra_integer(rid_int);
 
-      // Going through the elements in the mesh checking the pin_type_ids to axial
-      // region ids
-      if (_id_map.find(pt_id) == _id_map.end())
+      if (_pin_region_id_map.find(pt_id) != _pin_region_id_map.end())
       {
-        // element is in an assembly duct or background region since it doesn't have a pin type id
-        // that matches one in the map. What region it is in the periphery is then stored in the
-        // pt_id
-        dof_id_type at_id = elem->get_extra_integer(atid_int);
-        unsigned int peripheral_index = (UINT16_MAX - 1) - pt_id;
-        if (peripheral_index == 0)
-          // background region
-          elem->subdomain_id() = _background_id_map.find(at_id)->second[z_id];
-        else
-          // duct region
-          elem->subdomain_id() = _duct_id_map.find(at_id)->second[z_id][peripheral_index - 1];
+        // Pin type element, swap subdomains if necessary
+        const auto radial_idx = std::find(_pin_region_id_map[pt_id][0].begin(),
+                                          _pin_region_id_map[pt_id][0].end(),
+                                          base_rid) -
+                                _pin_region_id_map[pt_id][0].begin();
+        const auto elem_rid = _pin_region_id_map[pt_id][z_id][radial_idx];
+
+        // swap region ids if they are different
+        if (elem_rid != base_rid)
+        {
+          elem->set_extra_integer(rid_int, elem_rid);
+          bool has_block_names = !_pin_block_name_map[pt_id].empty();
+          auto elem_block_name = (has_block_names ? _pin_block_name_map[pt_id][z_id][radial_idx]
+                                                  : _block_name_prefix + std::to_string(elem_rid));
+          if (elem->type() == TRI3 || elem->type() == PRISM6)
+            elem_block_name += "_TRI";
+          const auto elem_block_id = getBlockId(elem_block_name, elem_rid);
+          elem->subdomain_id() = elem_block_id;
+          (*_build_mesh)->subdomain_name(elem_block_id) = elem_block_name;
+        }
       }
       else
       {
-        // region is in a pin so grab the different axial region ids and swap them
-        // since during extrusion all regions are given the same ID as the 2D layer
-        for (const auto i : make_range((_id_map.at(pt_id))[0].size()))
+        // element is in an assembly duct or background region since it doesn't
+        // have a pin type id that matches one in the map. Infer peripheral index
+        // from pin_type
+        dof_id_type at_id = elem->get_extra_integer(atid_int);
+        unsigned int peripheral_idx = (UINT16_MAX - 1) - pt_id;
+        bool is_background_region = peripheral_idx == 0;
+        const auto elem_rid =
+            (is_background_region ? _background_region_id_map[at_id][z_id]
+                                  : _duct_region_id_map[at_id][z_id][peripheral_idx - 1]);
+
+        // Swap region ids if they are different
```

Comment:
is this a swap/exchange? or more like a "change"
also explain why / give more details

same on the other 'swaps'

Reply:
These swaps occur if the block information at an axial level is different from the base 2-D information, since FancyExtruderMeshGenerator axially homogenizes the information when extruding from 2-D to 3-D. Therefore, we iterate through user defined axial variables and change them if necessary. "Change/update" is probably a better term

Reply:
ok let's have that infomration in the comment please

Reply:
More comments have been added to explain what is happening in these functions for PMG, AMG, and CMG